pkg/querier/worker/frontend_processor.go

```diff
@@ -121,6 +125,9 @@ func (fp *frontendProcessor) process(c frontendv1pb.Frontend_ProcessClient) erro
 }
 
 func (fp *frontendProcessor) runRequest(ctx context.Context, request *httpgrpc.HTTPRequest, statsEnabled bool, sendHTTPResponse func(response *httpgrpc.HTTPResponse, stats *stats.QueryStats) error) {
+	job := fp.memLimiter.NewJob()
+	defer job.Complete()
+	ctx = limiter.AddJobToContext(ctx, job)
```

Comment:
Is this too late? at this point u already pulled in the job from the queue right?

I think the intention is to not even pull the job from the QF/Query scheduler queue if "this" queries is in the yellow/red state?